docs/architecture/adr-038-state-listening.md

```diff
@@ -523,8 +812,21 @@ e.g.
             flush_timeout_ms = 5000 # Flush and wait for outstanding messages and requests to complete delivery when calling `StreamingService.Close(). (milliseconds)
             halt_app_on_delivery_error = true # Whether or not to halt the application when plugin fails to deliver message(s).
         ...
+=======
+# The plugin name used for streaming via gRPC
+plugin = "my_streaming_plugin"
```

Comment:
can there be multiple plugins at the same time?

Reply:
Apologies for the late reply.

Yes, there can be multiple plugins running at the same time. However, in this ADR proposal I restrict it to one plugin at a time. The reason are described below.

There are two ways we can expose multiple plugins:

1. we can expose any part of the system through plugins (ADR 038 is just one part). *This is what is currently desired*
2. we can expose multiple plugins through versioning `streaming.plugins = ["plugin_v1, plugin_v2, plugin_v3"`. *This is undesired*. Running multiple version would put an unnecessary burden on the node to stream the data out. A `fan-out` approach is better.

``` 
__________________________________     _____ SQL
|                         parsing |   /
|                            |    |  / 
|    SDK --->  grpc --->   plugin-|--------- NoSQL
|                            |    |  \
|                            |    |   \_____ Kafka
|                          local  |
|      local network       cache  |
-----------------------------------    
# this video explains the design of the go-plugin system.
# https://www.youtube.com/watch?v=SRvm3zQQc1Q           
```
@provenance-io we intend to go to Kafka and into other systems from there.

Come to think of it, perhaps it is better suited to change the config as below? It may better reflect the `go-plugin` design. Once the SDK team has further reviewed the design and #13472 implementation they may come to the same conclusion.

```toml
[streaming]
 
# ADR-038 plugin
[streaming.state-listening]
enable = true
keys = []
plugin = "abci_v1"


# Some other part of the system
[streaming.other1]
enable = true
plugin = "name_v1"

# Some other part of the system
[streaming.other2]
enable = true
plugin = "name_v1"
```
They can be loaded by calling `RegisterStreamingService` https://github.com/cosmos/cosmos-sdk/blob/db7840f9c3d345488c0faa0a446ec476bb7fac8f/simapp/app.go#L246